migrate.go

```diff
@@ -153,6 +153,11 @@ func collectMigrationsFS(fsys fs.FS, dirpath string, current, target int64) (Mig
 	if err != nil {
 		return nil, err
 	}
+
+	if len(sqlMigrationFiles)+len(registeredGoMigrations) == 0 {
+		return nil, fmt.Errorf("no migrations files in path: %s", dirpath)
```

Comment:
See https://github.com/pressly/goose/pull/362#issuecomment-1156477188

I suspect callers will want to account for this. We could make this:

1. Sentinel error value: `ErrNoMigrations = errors.New(...)`
2. A custom error: `struct NoMigrationsError {}`